### PR TITLE
Aims magnetic ordering

### PR DESCRIPTION
**Pull request by tpurcell90**
## Summary

Add magnetic ordering support for FHI-aims

## Additional dependencies introduced (if any)

* Will need an updated pymatgen version once the corresponding changes there are added

## Checklist

Work-in-progress pull requests are encouraged, but please put [WIP] in the pull request
title.

Before a pull request can be merged, the following items must be checked:

* [x] Code is in the [standard Python style](https://www.python.org/dev/peps/pep-0008/).
  The easiest way to handle this is to run the following in the **correct sequence** on
  your local machine. Start with running [`ruff`](https://docs.astral.sh/ruff) and `ruff format` on your new code. This will
  automatically reformat your code to PEP8 conventions and fix many linting issues.
* [x] Doc strings have been added in the [Numpy docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).
  Run [ruff](https://beta.ruff.rs/docs/rules/#pydocstyle-d) on your code.
* [x] Type annotations are **highly** encouraged. Run [mypy](http://mypy-lang.org) to
  type check your code.
* [x] Tests have been added for any new functionality or bug fixes.
* [x] All linting and tests pass.

Note that the CI system will run all the above checks. But it will be much more
efficient if you already fix most errors prior to submitting the PR. It is highly
recommended that you use the pre-commit hook provided in the repository. Simply run
`pre-commit install` and a check will be run prior to allowing commits.

### CODE REVIEW

**Comment by tpurcell90** [comment]
Tests fail because pymatgen dropped 3.9 support in the latest version

File: tests/ase/test_jobs.py [hunk 1]
```diff
@@ -45,6 +45,8 @@ def test_lennard_jones_static_maker(lj_fcc_ne_pars, fcc_ne_structure):
     assert output.output.energy == pytest.approx(-0.0179726955438795)
     assert output.structure.volume == pytest.approx(24.334)
     assert isinstance(output, AseStructureTaskDoc)
+
+    output.structure.properties = {}
```
Comment (tpurcell90) [inline tests/ase/test_jobs.py:49]: Seeing if this fixes the ASE jobs test

**Comment by tpurcell90** [comment]
All tests that fail are in cp2k or lobster. I think this is ready for review

File: src/atomate2/aims/schemas/calculation.py [hunk 2]
```diff
@@ -198,6 +219,8 @@ class Calculation(BaseModel):
         Whether FHI-aims completed the calculation successfully
     output: .CalculationOutput
         The FHI-aims calculation output
+    input: .CalculationOutput
```
Comment (JaGeo) [inline src/atomate2/aims/schemas/calculation.py:222]:
```suggestion
    input: .CalculationInput
```
?

**Comment by JaGeo** [comment]
I had a quick look. This looks very good to me. 

Maybe, you could also think about adding a bit more documentation on this part of the code in a future PR. Currently, people will mostly find the VASP workflows :).

**Comment by JaGeo** [comment]
@tpurcell90 Thanks! Will merge it as soon as the tests have run through!

**Comment by tpurcell90** [comment]
> Maybe, you could also think about adding a bit more documentation on this part of the code in a future PR. Currently, people will mostly find the VASP workflows :).

The FHI-aims team in general is starting to write some tutorials for using atomate2 with FHI-aims. I will refine upon those and add them here once I get the time to do all of that. Thanks for the quick review!

**Comment by JaGeo** [comment]
> > Maybe, you could also think about adding a bit more documentation on this part of the code in a future PR. Currently, people will mostly find the VASP workflows :).
> 
> The FHI-aims team in general is starting to write some tutorials for using atomate2 with FHI-aims. I will refine upon those and add them here once I get the time to do all of that. Thanks for the quick review!

That's awesome. I am looking forward to it. Thanks!

**Pull request merged**